node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -0,0 +1,583 @@
+          return callback(err);
+        }
+
+        // Return early if no activities were found.
+        if (_.isEmpty(activities)) {
+          return callback();
+        }
+
+        // 4. Calculate summary data for the week.
+        var activitySummary = summarizeActivities(activities, activityTypeConfiguration, users, userCount);
+        var courseData = {
+          'activitySummary': activitySummary,
+          'users': users
+        };
+
+        return callback(null, courseData);
+      });
+    });
+  });
+};
+
+/**
+ * Get recent activity data for a course.
+ *
+ * @param  {Context}    ctx                  Standard context containing the current user and the current course
+ * @param  {Object}     dateRange            Timestamp range for activity creation
+ * @param  {Date}       dateRange.start      Lower bound for activity creation timestamp
+ * @param  {Date}       dateRange.end        Upper bound for activity creation timestamp
+ * @param  {Function}   callback             Standard callback function
+ * @param  {Object}     callback.err         An error object, if any
+ * @param  {Activity[]} callback.activities  Activity data for the current course within the date range
+ * @api private
+ */
+var getActivitiesForDateRange = function(ctx, dateRange, callback) {
+  var options = {
+    'where': {
+      'course_id': ctx.course.id,
+      'created_at': {'gt': dateRange.start, 'lt': dateRange.end}
+    },
+    'include': [{
+      'model': DB.Asset,
+      'include': [
+        // All we need is the IDs of associated users, but Sequelize forces us to grab some user fields as well.
+        // Get basic fields only.
+        {
+          'model': DB.User, 
+          'as': 'users', 
+          'required': true, 
+          'attributes': UserConstants.BASIC_USER_FIELDS
+        },
+        // Get new comments for this week only, for purposes of counting keywords.
+        {
+          'model': DB.Comment, 
+          'required': true, 
+          'where': {
+            'created_at': {'gt': dateRange.start, 'lt': dateRange.end}
+          }
+        }
+      ]
+    }],
+    'limit': null,
+    'subQuery': false
+  };
+  DB.Activity.findAll(options).complete(function(err, activities) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Unable to get recent activities for a course.');
+      return callback(err);
+    }
+    return callback(null, activities);
+  });
+};
+
+
+/* Summary data calculation */
+
+
+/**
+ * Calculate summary data from a list of activities.
+ *
+ * @param  {Activity[]}   activities                   Activities within a date range for a given course
+ * @param  {Object}       activityTypeConfiguration    Activity type configuration for the course
+ * @param  {Object}       users                        All active users in the course indexed by id
+ * @param  {Number}       userCount                    Number of active users in the course
```

Comment:
Is it necessary to pass in the user count? You can just do something like `_.keys(users).length` when you need the length.